Change exception handling when saving the enrollment data of the validator

MichaelKim20 opened the pull request:
**This is PR to solve issue #888.**
When Nominator.valueExternalized occurs, the agreed data is recorded in the ledger.
There were occasional node shutdowns in this process.
I tried to find the cause through the log and code.
This is the cause of it. This is because when the enrollment data is abnormal, asset (0) is executed.

**Before changes**
When abnormal data is received, 'EnrollmentManager.addValidator' returns false then the node is terminated abnormally.

After changes
Abnormal data were filtered, and true returned.
Returns false only when it fails in the process of storing normal data.

Relates #888

MichaelKim20 commented:
I still need to make some more changes.

Geod24 commented on source/agora/consensus/data/Enrollment.d:
```diff
@@ -68,6 +68,14 @@ public struct Enrollment
     }
 }
 
+/// Results after adding the validator's enrollment data
+public enum ValidatorAddResult : ubyte
+{
+    Failure,    //  if an error occurs,
+    Invalid,    //  if the enrollment is abnormal,
+    Success     //  if the enrollment is added normally
+}
+
```
That's not a way we want to go. Having `Failure` and `Invalid` is quite confusing for the caller, and probably won't be handled correctly. It is also a [leaky abstraction](https://www.joelonsoftware.com/2002/11/11/the-law-of-leaky-abstractions/).

MichaelKim20 replied:
I think so too, I should change it again in a first way. it was complicated to modify the unit test.

MichaelKim20 committed:
Change exception handling when saving the enrollment of the validator

This returns `true` if the enrollment is abnormal,
if an error occurs, return `false`,
if added normally, return `true`
